server/src/main/java/org/elasticsearch/action/admin/indices/refresh/TransportShardRefreshAction.java

```diff
@@ -125,7 +125,6 @@ public void onPrimaryOperationComplete(
             IndexShardRoutingTable indexShardRoutingTable,
             ActionListener<Void> listener
         ) {
-            assert replicaRequest.primaryRefreshResult.refreshed() : "primary has not refreshed";
```

Comment:
I think this transitively asserted that RefreshResult is not UNKNOWN_PRIMARY_TERM, UNKNOWN_GENERATION which are negative, but I guess the readVLong and writeVLong of the request will assert that as well.

But maybe I'd feel easier if we asserted they're not UNKNOWN_PRIMARY_TERM, UNKNOWN_GENERATION

Reply:
That makes sense, I pushed 001c5551240892f96809750f7cc58a8bd53c2201